File: API/src/domain/entities/Calculator.ts
import Entity from "./Entity";
import { Status } from "../enums/Status";

interface CalculatorProps {
    id?: string;
    number1: number;
    number2: number;
    status: Status;
    result?: number;
    creationDateTime?: Date;
    finishDateTime?: Date;
}

export class Calculator extends Entity{
    id?: string;
    number1: number;
    number2: number;
    status: Status;
    result?: number;
    creationDateTime: Date;
    finishDateTime?: Date;

    constructor({id, number1, number2, status, result, creationDateTime, finishDateTime}: CalculatorProps){
        super();

        this.id = id;
        this.number1 = number1;
        this.number2 = number2;
        this.status = status;
        this.result = result;
        this.finishDateTime = finishDateTime;

        if(!creationDateTime)
            this.creationDateTime = new Date();
        else
            this.creationDateTime = creationDateTime;
    }

}

File: Worker/src/Worker.Domain/Entities/Calculator.cs
using MongoDB.Bson.Serialization.Attributes;
using System;
using Worker.Domain.Enums;

namespace Worker.Domain.Entities
{
    [BsonIgnoreExtraElements]
    public class Calculator: Entity
    {
        [BsonElement(elementName: "number1")]
        public double Number1 { get; private set; }

        [BsonElement(elementName: "number2")]
        public double Number2 { get; private set; }

        [BsonElement(elementName: "status")]
        public string Status { get; private set; }

        [BsonElement(elementName: "result")]
        public double? Result { get; private set; }

        [BsonElement(elementName: "creationDateTime")]
        public DateTime CreationDateTime { get; private set; }

        [BsonElement(elementName: "finishDateTime")]
        public DateTime? FinishDateTime { get; private set; }

        public Calculator(string id, double number1, double number2, Status status, DateTime creationDateTime): base(id)
        {
            Number1 = number1;
            Number2 = number2;
            Status = status.ToString();
            CreationDateTime = creationDateTime;
        }

        public void Sum()
        {
            Result = Number1 + Number2;
            Finish();
        }

        public void StartHandling()
        {
            this.Status = Worker.Domain.Enums.Status.Processing.ToString();
        }

        private void Finish()
        {
            FinishDateTime = DateTime.Now;
            Status = Worker.Domain.Enums.Status.Processed.ToString();
        }


    }
}
